external-crates/move/crates/move-compiler/src/typing/match_compilation.rs

```diff
@@ -2676,6 +2694,186 @@ fn find_counterexample_impl(
+
+            let suggestion = if is_positional {
+                PS::UnpackPositionalStruct {
+                    module: mident,
+                    name,
+                    field_count: fields.len(),
+                }
+            } else {
+                PS::UnpackNamedStruct {
+                    module: mident,
+                    name,
+                    fields: fields.into_iter().map(|(field, _)| field.value()).collect(),
+                }
+            };
+            let info = MissingMatchArmsInfo {
+                arms: vec![suggestion],
+            };
+            context
+                .env
+                .add_ide_annotation(loc, IDEAnnotation::MissingMatchArms(Box::new(info)));
+        } else {
+            // If there's a default arm, no suggestion is necessary.
+            if matrix.has_default_arm() {
+                return;
+            }
+
+            let mut unmatched_variants = context
+                .enum_variants(&mident, &name)
+                .into_iter()
+                .collect::<BTreeSet<_>>();
+            let ctors = matrix.first_variant_ctors();
+            for ctor in ctors.keys() {
+                unmatched_variants.remove(ctor);
+            }
+            // If all of the variants were matched, no suggestion is necessary.
+            if unmatched_variants.is_empty() {
+                return;
+            }
+            let mut arms = vec![];
+            for variant in unmatched_variants {
+                let is_empty = context.enum_variant_is_empty(&mident, &name, &variant);
+                let is_positional = context.enum_variant_is_positional(&mident, &name, &variant);
+                let Some(fields) = context.enum_variant_fields(&mident, &name, &variant) else {
+                    context.env.add_diag(ice!((
+                        loc,
+                        "Tried to look up fields for this enum and found none"
+                    )));
+                    continue;
+                };
+                let suggestion = if is_empty {
+                    PS::UnpackEmptyVariant {
+                        module: mident,
+                        enum_name: name,
+                        variant_name: variant,
+                    }
+                } else if is_positional {
+                    PS::UnpackPositionalVariant {
+                        module: mident,
+                        enum_name: name,
+                        variant_name: variant,
+                        field_count: fields.len(),
+                    }
+                } else {
+                    PS::UnpackNamedVariant {
+                        module: mident,
+                        enum_name: name,
+                        variant_name: variant,
+                        fields: fields.into_iter().map(|(field, _)| field.value()).collect(),
+                    }
+                };
+                arms.push(suggestion);
+            }
+            let info = MissingMatchArmsInfo { arms };
```

Comment:
The absolute nittiest of nits, but I know it would drive me crazy: can we return the suggestions for variants in their declaration order?

Reply:
Change made before I looked at your feedback :)